tests/php/src/PairedRoutingTest.php

```diff
@@ -429,15 +428,53 @@ public function test_filter_unique_post_slug( $post_name, $other_existing_post_n
 		$this->assertSame( $expected_slug, $actual_slug );
 	}
 
-	/** @covers ::add_paired_request_hooks() */
-	public function test_add_paired_request_hooks_when_does_have_endpoint() {
+	/** @return array */
+	public function get_data_for_test_add_paired_request_hooks_when_does_have_endpoint() {
+		return [
+			'query_var'           => [
+				Option::PAIRED_URL_STRUCTURE_QUERY_VAR,
+				false,
+			],
+			'legacy_transitional' => [
+				Option::PAIRED_URL_STRUCTURE_LEGACY_TRANSITIONAL,
+				false,
+			],
+			'path_suffix'         => [
+				Option::PAIRED_URL_STRUCTURE_PATH_SUFFIX,
+				true,
+			],
+			'legacy_reader'       => [
+				Option::PAIRED_URL_STRUCTURE_LEGACY_READER,
+				true,
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider get_data_for_test_add_paired_request_hooks_when_does_have_endpoint()
+	 * @covers ::add_paired_request_hooks()
+	 * @covers ::is_using_path_suffix()
+	 * @param string $structure
+	 * @param bool   $using_path_suffix
+	 */
+	public function test_add_paired_request_hooks_when_does_have_endpoint( $structure, $using_path_suffix ) {
+		$this->set_permalink_structure( '/%year%/%monthnum%/%day%/%postname%/' );
+		AMP_Options_Manager::update_option( Option::PAIRED_URL_STRUCTURE, $structure );
+		$this->assertEquals( $using_path_suffix, $this->instance->is_using_path_suffix() );
 		$post = self::factory()->post->create();
+		remove_all_actions( 'wp' );
 		$this->go_to( amp_get_permalink( $post ) );
-		$this->instance->add_paired_request_hooks();
 		$this->assertTrue( $this->instance->has_endpoint() );
+		$this->instance->add_paired_request_hooks();
 		$this->assertEquals( 1000, has_filter( 'old_slug_redirect_url', [ $this->instance, 'maybe_add_paired_endpoint' ] ) );
 		$this->assertEquals( 1000, has_filter( 'redirect_canonical', [ $this->instance, 'maybe_add_paired_endpoint' ] ) );
+		if ( $using_path_suffix ) {
+			$this->assertEquals( 0, has_filter( 'get_pagenum_link', [ $this->instance, 'filter_get_pagenum_link' ] ) );
+		} else {
+			$this->assertFalse( has_filter( 'get_pagenum_link', [ $this->instance, 'filter_get_pagenum_link' ] ) );
+		}
 		$this->assertFalse( has_action( 'wp_head', 'amp_add_amphtml_link' ) );
+		$this->assertEquals( $using_path_suffix, $this->instance->is_using_path_suffix() );
```

Comment:
This check was already done on line 463. Is it necessary once more?

Reply:
Ah no. It was redundant. Removed in c467d11c6.